packages/experience-builder-sdk/src/blocks/editor/VisualEditorContext.tsx

```diff
@@ -116,28 +115,35 @@ export function VisualEditorContextProvider({
     );
   }, [entityStore, locale]);
 
+  // Either gets called when dataSource changed or designComponetsRegistry changed (manually)
+  const fetchMissingEntities = useCallback(async () => {
+    const entityLinks = [...Object.values(dataSource), ...designComponentsRegistry.values()];
+    const { missingAssetIds, missingEntryIds } = entityStore.getMissingEntityIds(entityLinks);
+    // Only continue and trigger rerendering when we need to fetch something and we're not fetching yet
+    if (!missingAssetIds.length && !missingEntryIds.length) return;
```

Comment:
```suggestion
    if (!missingAssetIds.length && !missingEntryIds.length) {
      setInitialEntitiesFetched(true);
      return;
    }
```
I got a blank canvas when trying this out and the reason seemed to be that there were no missing entities but `initialEntitiesFetched` was still false. This seems to have fixed it. In my case there were no missing entities because they had been fetched in preview mode already.

Reply:
Aaaaah, sorry I was too quick again on this one. Will test this carefully again as well. Thanks for showing me this fix!